js/script.js

```diff
@@ -89,14 +90,15 @@ $(function () {
     // and instead the minimize buttons should be used.
     //setupFadeoutOnIdle();
 
-    var getNextSlideIndex = function(currentIndex) {
-        if(!rp.settings.nsfw) {
+    var getNextSlideIndex = function (currentIndex) {
+        if (!rp.settings.nsfw) {
             // Skip any nsfw if you should
-            for(var i = currentIndex + 1; i < rp.photos.length; i++) {
+            for (var i = currentIndex + 1; i < rp.photos.length; i++) {
                 if (!rp.photos[i].over18) {
                     return i;
                 }
             }
+            return 0;
```

Comment:
here not handling situation where none of available pictures is non-nsfw caused ajax looping, but it's obscure usecase